From 50b5545d274e4182696dd9d1634aa1379a9ce172 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 29 Nov 2024 18:25:51 +0100
Subject: [PATCH 1517/1695] plat-stm32mp1: simplify
 print_ccm_decryption_duration

Simplify call of print_ccm_decryption_duration to avoid #if
and __maybe_unused in code.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 706560db02161e57978279d97e02fe21686cb9c3 plat-stm32mp1: simplify print_ccm_decryption_duration]
Fixes: cf483e4784e5 ("plat-stm32mp1: add low power drivers for stm32mp1")
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/423279
Change-Id: I2ca53b38a42e9da54c2dc4becba3447564e5ff46
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/437075
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp1/pm/context.c | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/pm/context.c b/core/arch/arm/plat-stm32mp1/pm/context.c
index cfc91404b..e48e8e469 100644
--- a/core/arch/arm/plat-stm32mp1/pm/context.c
+++ b/core/arch/arm/plat-stm32mp1/pm/context.c
@@ -171,9 +171,9 @@ static void save_time(void)
 		panic();
 }
 
-#if TRACE_LEVEL >= TRACE_DEBUG
-static void __maybe_unused print_ccm_decryption_duration(void)
+static void print_ccm_decryption_duration(void)
 {
+#if defined(CFG_STM32MP1_OPTEE_IN_SYSRAM) && TRACE_LEVEL >= TRACE_DEBUG
 	vaddr_t stgen = stm32mp_stgen_base();
 	struct retram_resume_ctx *ctx = get_retram_resume_ctx();
 
@@ -184,12 +184,8 @@ static void __maybe_unused print_ccm_decryption_duration(void)
 	     io_read32(stgen + CNTFID_OFFSET));
 
 	clk_disable(pm_clocks.bkpsram);
-}
-#else
-static void __maybe_unused print_ccm_decryption_duration(void)
-{
-}
 #endif
+}
 
 static void restore_time(void)
 {
@@ -216,9 +212,7 @@ static void restore_time(void)
 	/* Balance clock enable(RTC) at save_time() */
 	clk_disable(pm_clocks.rtc);
 
-#ifdef CFG_STM32MP1_OPTEE_IN_SYSRAM
 	print_ccm_decryption_duration();
-#endif
 }
 
 /* Clear the content of the PM mailbox */
-- 
2.47.2

